### PR TITLE
Add deployment unit tests

### PR DESCRIPTION
**Pull request by sebrandon1**
Resolves: #270 

- Adds unit tests for the deployment package.  
- Modifies the interface used in the `Builder`.

### CODE REVIEW

File: pkg/deployment/deployment.go [hunk 1]
```diff
@@ -118,7 +124,9 @@ func Pull(apiClient *clients.Settings, name, nsname string) (*Builder, error) {
 
 // WithNodeSelector applies a nodeSelector to the deployment definition.
 func (builder *Builder) WithNodeSelector(selector map[string]string) *Builder {
-	if valid, _ := builder.validate(); !valid {
+	if valid, err := builder.validate(); !valid {
+		builder.errorMsg = err.Error()
```
Comment (trewest) [inline pkg/deployment/deployment.go:128]: I think it is redundant. builder.errorMsg should already contain what you are registering in err

Comment (sebrandon1) [reply pkg/deployment/deployment.go:128]: I'll double-check what I was trying to do with this one.

File: pkg/deployment/deployment_test.go [hunk 2]
```diff
@@ -431,3 +451,249 @@ func TestWithToleration(t *testing.T) {
...
+
+	for _, testCase := range testCases {
+		var runtimeObjects []runtime.Object
+
+		if testCase.deploymentExistsAlready {
+			runtimeObjects = append(runtimeObjects, generateTestDeployment())
+		}
+
+		testBuilder := buildTestBuilderWithFakeObjects(runtimeObjects)
+		result, err := testBuilder.Update()
```
Comment (kononovn) [inline pkg/deployment/deployment_test.go:519]: Can you create test which actually mutate some filed using update function ?

**Review by kononovn — CHANGES_REQUESTED** [review]
Overall looks good. Just 1 comment to fix

File: pkg/deployment/deployment.go (hunk 1 above)
Comment (sebrandon1) [reply pkg/deployment/deployment.go:128]: I double checked and removed it.  Good catch I'm not sure why I added that but it wasn't needed.  😄

**Review by kononovn — APPROVED** [review]
/lgtm

**Pull request merged**